independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -0,0 +1,504 @@
+        //as MP has super broken class loading where it attempts to resolve stuff from the parent
+        //will hopefully be fixed in 1.4
+        //e.g. https://github.com/eclipse/microprofile-config/issues/390
+        //e.g. https://github.com/eclipse/microprofile-reactive-streams-operators/pull/130
+        super(null);
+        this.name = builder.name;
+        this.elements = builder.elements;
+        this.bytecodeTransformers = builder.bytecodeTransformers;
+        this.bannedElements = builder.bannedElements;
+        this.parentFirstElements = builder.parentFirstElements;
+        this.parent = builder.parent;
+        this.parentFirst = builder.parentFirst;
+        this.resettableElement = builder.resettableElement;
+        this.transformerClassLoader = builder.transformerClassLoader;
+        this.aggregateParentResources = builder.aggregateParentResources;
+    }
+
+    public static Builder builder(String name, ClassLoader parent, boolean parentFirst) {
+        return new Builder(name, parent, parentFirst);
+    }
+
+    private String sanitizeName(String name) {
+        if (name.startsWith("/")) {
+            return name.substring(1);
+        }
+        return name;
+    }
+
+    private boolean parentFirst(String name, ClassLoaderState state) {
+        return parentFirst || state.parentFirstResources.contains(name);
+    }
+
+    public void reset(Map<String, byte[]> resources,
+            Map<String, List<BiFunction<String, ClassVisitor, ClassVisitor>>> bytecodeTransformers, ClassLoader transformerClassLoader) {
+        if (resettableElement == null) {
+            throw new IllegalStateException("Classloader is no resettable");
+        }
+        this.transformerClassLoader = transformerClassLoader;
+        synchronized (this) {
+            resettableElement.reset(resources);
+            this.bytecodeTransformers = bytecodeTransformers;
+            state = null;
+        }
+    }
+
+    @Override
+    public Enumeration<URL> getResources(String nm) throws IOException {
+        ClassLoaderState state = getState();
+        String name = sanitizeName(nm);
+        //for resources banned means that we don't delegate to the parent, as there can be multiple resources
+        //for single resources we still respect this
+        boolean banned = state.bannedResources.contains(name);
+        List<URL> resources = new ArrayList<>();
+        //ClassPathElement[] providers = loadableResources.get(name);
+        //if (providers != null) {
+        //    for (ClassPathElement element : providers) {
+        //        resources.add(element.getResource(nm).getUrl());
+        //    }
+        //}
+        for (ClassPathElement i : elements) {
+            ClassPathResource res = i.getResource(nm);
+            if (res != null) {
+                resources.add(res.getUrl());
+            }
+        }
+        if (!banned) {
+            if(resources.isEmpty() || aggregateParentResources) {
```

Comment:
This one looks somewhat confusing. If `aggregateParentResources` is false then it will still be looking for the resources in the parent but only if nothing was found locally. Is there a specific use-case where this is needed? I'm just thinking wouldn't simply consulting the parent only if `aggregateParentResources` was true work?

Reply:
this is used for the deployment and runtime class loaders, so that getResources gets everything that would be on the actual application class path, without anything that may have leaked in on the system class path.

I am not sure about just never consulting the parent, as it would effectively hide JDK resources. Arc needs to be able to index JDK classes at times, so if we did not delegate at all this would cause problems.